mgmt/api/APITestCliRemote.cc

```diff
@@ -626,7 +626,9 @@ print_remap_ele(TSRemapEle *ele)
   }
   // from port
   if (ele->from_port != TS_INVALID_PORT) {
-    snprintf(buf, sizeof(buf), "%s:%d", buf, ele->from_port);
+    char port_buf[16];
+    snprintf(port_buf, sizeof(port_buf), ":%d", ele->from_port);
+    ink_strlcat(buf, port_buf, sizeof(buf));
```

Comment:
int used = strlen(buf);
snprintf(buf + used, sizeof(buf) - used, ":%d", ele->from_port);

Reply:
Although gcc8 may think it's ok, isn't that share the last byte (a byte for null terminator)?

Reply:
I don't understand what you mean.  strlen() does not include the nul character in the length of the string.

Reply:
Maybe the best thing would be to use FixedBufferWriter.

Reply:
I meant `buf[strlen(buf)]` is apart of the original string and it will be overwritten by snprintf(). So, snprintf() technically breaks the original string, of course it works perfectly though.

Probably your code is fine and better than mine because snprintf() doesn't find the end of the original string during its process and there's no need to write port twice. I think I misread the error message from gcc8.